### PR TITLE
Packed all dlls into the binary

### PR DESCRIPTION
**Pull request by rishav394**
## Description

Used `Fody` to pack all the dlls into to main `Titan.exe` binary. 

## Type

```
Feature
```

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) guide.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I tested it on **Windows** and it's working as intended.

## Discussion

Used Costura.Fody 2.0.1 from https://www.nuget.org/packages/Costura.Fody/
Not supported by mono.
Total size of the assembly is not affected as Costura has a built in compression mechanism.

### CODE REVIEW

**Comment by rishav394** [comment]
xD
I didn't request a review from @Marc3842h either. I guess its a default thing (set to owner only maybe and not collaborator).

File: Titan/Titan.csproj [hunk 1]
```diff
@@ -277,6 +280,8 @@
   <Import Project="..\packages\Eto.Platform.Mac.2.4.1\build\Eto.Platform.Mac.targets" Condition="Exists('..\packages\Eto.Platform.Mac.2.4.1\build\Eto.Platform.Mac.targets')" />
   <Import Project="..\packages\Eto.Platform.Mac64.2.4.1\build\Eto.Platform.Mac64.targets" Condition="Exists('..\packages\Eto.Platform.Mac64.2.4.1\build\Eto.Platform.Mac64.targets')" />
   <Import Project="..\packages\Mono.Posix.5.4.0.201\build\net45\Mono.Posix.targets" Condition="Exists('..\packages\Mono.Posix.5.4.0.201\build\net45\Mono.Posix.targets')" />
+  <Import Project="..\packages\Fody.3.0.3\build\Fody.targets" Condition="Exists('..\packages\Fody.3.0.3\build\Fody.targets')" />
+  <Import Project="..\packages\Costura.Fody.2.0.1\build\Costura.Fody.targets" Condition="Exists('..\packages\Costura.Fody.2.0.1\build\Costura.Fody.targets')" />
```
Comment (Marc3842h) [inline Titan/Titan.csproj]: I'd say we add the condition that `$(OS)` is not `Unix` (macOS and Linux) as this isn't supported on Mono.

File: Titan/FodyWeavers.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Weavers>
+  <Costura/>
```
Comment (Marc3842h) [inline Titan/FodyWeavers.xml]: Maybe add the `NoCleanup` option for MacOS and Linux here? May be covered by change above if I read the docs correctly.

File: TitanTest/TitanTest.csproj [hunk 3]
```diff
@@ -130,4 +131,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
+</Project>
```
Comment (Marc3842h) [inline TitanTest/TitanTest.csproj]: Please add a newline here.

File: TitanTest/app.config [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <assemblyIdentity name="xunit.execution.desktop" publicKeyToken="8d05b1bb7a6fdb6c" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-2.3.0.3820" newVersion="2.3.0.3820" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-11.0.0.0" newVersion="11.0.0.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (Marc3842h) [inline TitanTest/app.config]: Please use 4 space indentation. Also add a newline here.

**Comment by Marc3842h** [comment]
@raspbianlike A review for this was automatically requested as I'm a code owner of this repository. You are a coder owner of the `Titan/Managers/ThreadManager.cs` file as that contains the only major feature you contributed. Feel free to read more about it [on the CodeOwners file](https://github.com/Marc3842h/Titan/blob/master/.github/CODEOWNERS).

File: Titan/FodyWeavers.xml (hunk 2 above)
Comment (rishav394) [reply Titan/FodyWeavers.xml]: Yea I dont think we have to change this if OS is not windows. The file is rendered useless anyways if Fody is not used at all.

**Comment by raspbianlike** [comment]
shaddefuckap i will cod b1g ok? @shiho vouch

**Comment by Shihobiggei** [comment]
vOuched

**Comment by Marc3842h** [comment]
Could you address the 4 space indentation in the `.xml` and `.config` files? Then I'll merge this.

**Comment by Marc3842h** [comment]
Thank you!

**Pull request merged**